### ISSUE TITLE
mgmt Network, support mandatory `priority` in RequestRoutingRules

### ISSUE THREAD

**Issue by XiaofeiCao** [Azure/azure-sdk-for-java#28899]
From api-version `2021-08-01` and up, `priority` is a required property in `RequestRoutingRules`.

**Comment by XiaofeiCao** [comment]
Due to the fact that for some SKUs, `priority` is mandatory and for others, it's forbidden, we don't want `withPriority` to become a mandatory stage on rule creation. We want to provide default values for them.

We can do the following:

1. From SDK side, let 10010-20000 be the auto-assigned priority range. In javadoc, recommend user to self assign using 1-10000. （auto-assigned will be after user-assigned by default)
2. Future auto-assigned priority will enumerate from 10010 by the interval of 10 (10010, 10020, 10030, ..., etc.) If it comes across a duplicate priority (user-assigned by accident), skip that one and continue the next. If no available priority exists, throw an exception(or ignore that rule).

@weidongxu-microsoft What's your thoughts on this strategy?

**Comment by weidongxu-microsoft** [comment]
Sound good.

In 2, is there another possible conflict that could also happen when Update an existing Gateway (previously no priority), and now service return 10010, 10020, etc.?
(handling should be same, just want to be clear about the cause)

Exception seems to be better (should be extremely rare?).

**Comment by XiaofeiCao** [comment]
Yes, this is the hard case and the same for updating existing gateway with auto-assigned priority by SDK... 

For this case, suppose we can add a warning in javadoc like: 
> For updating the priority for a rule that you didn't assigned priority before, we recommend updating all existing rules for that Gateway altogether. Otherwise, it will lead to unexpected ordering.

**Comment by weidongxu-microsoft** [comment]
Yeah, there seems not much we can do here...

**Comment by XiaofeiCao** [comment]
Fixed by https://github.com/Azure/azure-sdk-for-java/pull/29503
